### ISSUE TITLE
Nettle API breakage

### ISSUE THREAD

**Issue by staticfloat** [randyzwitch/OAuth.jl#7]
[Nettle](https://github.com/staticfloat/Nettle.jl) has just broken its API significantly, in order to be much more precompile-friendly.  The big takeaway is that methods such as `md5_hash(data)` should be mapped to `digest("md5", data)`, there are no more auto-generated types for hashes, and internally the code is a little easier to grok.  This new version of `Nettle.jl` has not been released yet, I will make every effort to ensure that dependent packages such as this one have a chance to adapt before releasing the new version.

**Comment by randyzwitch** [comment]
Thanks for the heads-up; whenever you've finalized your changes, I'll incorporate them here. Luckily, this package is pretty small.

**Comment by staticfloat** [comment]
The changes are finalized, feel free to change your package whenever you are ready.  :)

**Comment by randyzwitch** [comment]
What do I need to do, work off of master, then specify a version number in REQUIRES?

**Comment by staticfloat** [comment]
Precisely.

**Comment by randyzwitch** [comment]
Maybe I'm missing something, but did you change the API for the code usage I do? The README documentation appears to be the same.

https://github.com/randyzwitch/OAuth.jl/blob/master/src/OAuth.jl#L38-L43

https://github.com/randyzwitch/OAuth.jl/blob/master/src/OAuth.jl#L109-L113

**Comment by staticfloat** [comment]
In your case, you need to change `HashState` to `Hasher`, and I think that's it!  Super simple!

**Comment by staticfloat** [comment]
Oh, one other thing; `SHA1` needs to become `"SHA1"`.  Notice the quotes; there aren't per-hash types anymore; we do things with strings now.

**Comment by randyzwitch** [comment]
Ah, staring at code too long today already, I missed the quotes!

**Comment by staticfloat** [comment]
Once https://github.com/JuliaLang/METADATA.jl/pull/3537 is merged, this package's Nettle functionality will cease to work.  I've opened a pull request to this repository with the necessary fixes.

**Comment by randyzwitch** [comment]
There was an error during the PR checks, is that because JuliaLang/METADATA.jl#3537 hasn't been merged?

**Comment by staticfloat** [comment]
Yes, exactly. Once you merge into your master branch, I'll release Nettle
v0.2.0, then you can ensure everything works on master and release a new
version.
